torch/_dynamo/guards.py

```diff
@@ -866,13 +904,55 @@ def guard_error_hook(
     # column number of which subexpression failed.  But that would also
     # require us to have the TRUE code that was eval'ed, not a shoddy
     # reconstruction (like is done here)
-    print("lambda " + ", ".join(guard_fn.args) + ":")
-    print(" ", " and\n  ".join(guard_fn.code_parts))
+    print(make_guard_fn(guard_fn.code_parts, {}, []))
 
 
 set_guard_error_hook(guard_error_hook)
 
 
+def make_guard_fn(code_parts, closure_vars, part_list):
+    # TODO(voz): Move this somewhere more general so we don't have to violate heirarchy?
+    from torch._inductor.utils import IndentedBuffer
+
+    make_fail_buf = IndentedBuffer()
+    make_fail_buf.writeline("def __fail(code_part_idx):")
+    with make_fail_buf.indent():
+        make_fail_buf.writeline("code_part = part_list[code_part_idx]")
+        make_fail_buf.writeline("code_part.scope = locals()")
+        make_fail_buf.writeline("code_part.scope['L'] = L")
```

Comment:
I guess I still don't really understand why we have to store the locals on the code_part. Can  you even call this a code_part now.

Reply:
Is this so guard fail fn can reexecute the guard code? But is this really necessary now that you have code_part instrumentation?

Reply:
Yea, we still need to do this.

Reply:
Tell me more?